components/server/src/raft_engine_switch.rs

```diff
@@ -45,8 +45,11 @@ fn rename_to_tmp_dir<P1: AsRef<Path>, P2: AsRef<Path>>(src: P1, dst: P2) {
 fn clear_raft_engine(engine: &RaftLogEngine) -> Result<(), EngineError> {
     let mut batch_to_clean = engine.log_batch(0);
     for id in engine.raft_groups() {
-        let state = engine.get_raft_state(id)?.unwrap();
-        engine.clean(id, &state, &mut batch_to_clean)?;
+        let raft_state = engine.get_raft_state(id)?.unwrap();
+        let first_index = engine.first_index(id).unwrap_or(0);
+        let mut apply_state = RaftApplyState::default();
+        apply_state.mut_truncated_state().set_index(first_index);
```

Comment:
`truncated_state.index` was set as `first_index` and added 1 [here](https://github.com/tikv/tikv/pull/11059/files#diff-455d4f08e0767d97cc39ae6cde4ce31e163cde02e52e91a700ae9d01065f9b69R159), is it right?